aikau/src/main/resources/webscript-libs/service-filtering.lib.js

```diff
@@ -1,17 +1,21 @@
 // This library file provides a convenient way in which services provided from multiple libraries can be filtered 
 // against each other to ensure that there are no duplicates (e.g. you want to avoid having two DialogServices on a
 // page otherwise you'll get two dialogs for each request!).
-function alfAddUniqueServices(currentServices, servicesToFilter) {
-   // Add the current services into the array that will ultimately be returned...
-   // Unique services not already defined (or that are configured with a different pubSubScope
-   // will be added) as well...
-   var filteredServices = [].concat(currentServices);
-
+function alfAddUniqueServices(currentServices, servicesToFilter, includeCurrentServices) {
+   var filteredServices = [];
+   
    // There's only filtering to do with we have been provided with some services to
    // filter against an array of current services...
    if (servicesToFilter && servicesToFilter.length &&
        currentServices && currentServices.length)
    {
+      // It is possible to request to NOT include the currentServices in the filtered result,
+      // this allows multiple Surf Components using Aikau on a page to avoid duplicating each other
+      if (includeCurrentServices !== false)
```

Comment:
This seems like an unusual way of doing it. Having an optional argument that, when supplied, must be false to take effect? The more normal way would be to call it something like `excludeCurrentServices` and to then pass through a truthy value when that non-default action is required. Is there a specific reason that it's been done this way round?